app/assets/javascripts/submissions.coffee

```diff
@@ -16,50 +16,76 @@ plantLineSelectOptions = ->
   templateResult: formatPlantLine
   templateSelection: formatPlantLine
 
+plantLineListSelectOptions = ->
+  $.extend({}, plantLineSelectOptions(), multiple: true)
+
+plantLineGeneticStatusSelectOptions = ->
+  multiple: true
+  tags: true
+  maximumSelectionLength: 1
+
 defaultSelectOptions = ->
   allowClear: true
 
-buttonTargets =
-  'add-plant-line': '.new-plant-line'
-  'select-existing-plant-line': '.existing-plant-line'
-  'select-existing-taxonomy-term': '.existing-taxonomy-term'
-
-buttonTargetSelectOptions =
-  'add-plant-line': plantLineSelectOptions
-  'select-existing-plant-line': plantLineSelectOptions
-  'select-existing-taxonomy-term': defaultSelectOptions
+appendToSelectedPlantLineLists = (data) ->
+  $select = $('.plant-line-list')
+  selectedValues = $select.val() || []
 
-markActiveButton = (button) ->
-  $(button).parent().find('button').removeClass('active')
-  $(button).addClass('active')
+  $option = $('<option></option>').attr(value: data.plant_line_name).text(data.plant_line_name)
+  $select.append($option)
 
-activateTargetFields = (button) ->
-  target = buttonTargets[button.id]
-  initialized = !$(target).hasClass('hidden')
+  selectedValues.push(data.plant_line_name)
+  $select.val(selectedValues)
+  $select.trigger('change') # required to notify select2 about changes, see https://github.com/select2/select2/issues/3057
 
-  $('.new-plant-line, .existing-plant-line, .existing-taxonomy-term').not(target).hide()
-  $(target).removeClass('hidden').show()
+  # add all PL attributes to DOM so it can be sent with form
+  $form = $('.edit_submission')
+  $container = $('<div></div').attr(id: newPlantLineForListContainerId(data.plant_line_name))
+  $container.appendTo($form)
 
-  unless initialized
-    $(target).find('select').select2(buttonTargetSelectOptions[button.id]())
+  $.each(data, (attr, val) ->
+    $input = $("<input type='hidden' name='submission[content][new_plant_lines][][" + attr + "]' />")
+    $input.val(val)
+    $input.appendTo($container)
+  )
 
-switchTarget = (button) ->
-  markActiveButton(button)
-  activateTargetFields(button)
+newPlantLineForListContainerId = (plant_line_name) ->
+  'new-plant-line-' + plant_line_name.split(/\s+/).join('-').toLowerCase()
```

Comment:
Are that complex processing required due to not having a proper integer id for a PL? This is why we need to rely on plant_line_name for identification?